answerking_app/mixins/OrderItemMixins.py

```diff
@@ -0,0 +1,137 @@
+from django.db import DataError
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.utils.serializer_helpers import ReturnDict
+
+from answerking_app.models.models import Order, Item
+from answerking_app.models.serializers import (
+    OrderSerializer,
+    OrderLineSerializer,
+)
+from answerking_app.views.ErrorType import ErrorMessage
+
+
+class OrderItemUpdateMixin:
+    def update(self, request, *args, **kwargs) -> Response | None:
+        partial: bool = kwargs.pop("partial", False)
+
+        order_id: str = kwargs.get("order_id")
+        item_id: str = kwargs.get("item_id")
+
+        order: Order | None = Order.objects.get(pk=order_id)
+        item: Item | None = Item.objects.get(pk=item_id)
+
+        if not order or not item:
+            error_msg: ErrorMessage = {
+                "error": {
+                    "message": "Request failed",
+                    "details": "Object not found",
+                }
+            }
+            return Response(
+                error_msg,
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        body: dict = request.data
+        quantity: int = body["quantity"]
+        serialized_body: OrderLineSerializer = OrderLineSerializer(data=body)
+        serialized_body.is_valid(raise_exception=True)
+
+        if quantity == 0:
+            order.order_items.remove(item)
+        elif item not in order.order_items.all():
+            try:
+                order.order_items.add(
+                    item,
+                    through_defaults={
+                        "quantity": quantity,
+                        "sub_total": item.price * quantity,
+                    },
+                )
+            except DataError:
+                return None
+        else:
+            try:
+                order.orderline_set.filter(item=item.id).update(
+                    quantity=quantity, sub_total=quantity * item.price
+                )
+            except DataError:
+                return None
+
+        serializer = self.get_serializer(
+            order, data=request.data, partial=partial
+        )
```

Comment:
seen as you have imported OrderSerializer it seems redundant to get it again. Wouldn't `serialized_data = OrderSerializer(data=request.data)` work here?